datafusion/physical-plan/src/aggregates/no_grouping.rs

```diff
@@ -77,6 +77,11 @@ impl AggregateStream {
         let baseline_metrics = BaselineMetrics::new(&agg.metrics, partition);
         let input = agg.input.execute(partition, Arc::clone(&context))?;
 
+        // Only wrap no‐grouping aggregates in our YieldStream
```

Comment:
In my own testing with `partition_count = 1` group by aggregates suffer from the same problem

Reply:
Thank you @pepijnve for review, i will try to reproduce it for group by aggregates.

Reply:
Thank you @pepijnve , i also added the grouping support in latest PR.

Reply:
From testing result, it seems the grouping by cases have some performance regression.